Added normalized scalar for canvas drawing, added Box drawing flags, moved target string inside Render class

64blit opened the pull request:
Additionally, a "dev" script in the Renderer2d that adds a build system on .ts file changes, as well as a node/vite server that refreshes on any live changes, allowing for quick testing.

` cd src/eyepop-render-2d; npm run dev `
`cd example/vite; npm run dev` > http://localhost:5173/


-------------
Squashed commits:

[update] Moved target string inside Renderer interface and added dynamic canvas scalar

[update] added unreasonably large file testing

[fix] added back rect width based scaling, to help with small bounds

[fix] added sensible hdpi scaling

[fix] normalized width of lines

[update] Added normalized font scaling

[update] Added dev script to build on any file changes

[fix] fixed text offset for multiple nested labels

[update] Added drawing flags for finer control

[update] Moved target string inside Renderer interface and added dynamic canvas scalar to handle large media

[update] Added scale factor to account for large images

[update] moved target string into Render interface

[create] added larger test file

[update] Moved target string inside Renderer interface and added dynamic canvas scalar

tschulz commented on examples/vite/.gitignore:
Why do we need another .gitignore in a subfolder?

64blit replied:
Indeed! File has been deleted https://github.com/eyepop-ai/eyepop-sdk-node/pull/31/commits/512452a08ed3e663cf54e446e8d789c4c2cd4d8c

tschulz commented on examples/vite/public/example.jpg:
```diff
@@ -0,0 +1 @@
+/Users/edmunddao/_SPACE/EyePop.ai/Untitled/eyepop-sdk-node/examples/example.jpg
```
This seems to be an erroneous commit.

64blit replied:
Also a sym link, resolved via [1b14f62](https://github.com/eyepop-ai/eyepop-sdk-node/pull/31/commits/1b14f6251cf82dc803abbcdb14a2710bb675b917)

tschulz commented on examples/vite/public/large_example.jpg:
```diff
@@ -0,0 +1 @@
+/Users/edmunddao/_SPACE/EyePop.ai/Untitled/eyepop-sdk-node/examples/large_example.jpg
```
ditto

tschulz commented on examples/vite/src/eyepop.min.js:
```diff
@@ -0,0 +1 @@
+/Users/edmunddao/_SPACE/EyePop.ai/Untitled/eyepop-sdk-node/src/eyepop/dist/eyepop.min.js
```
what are those files?

64blit replied:
They were symlinks used for testing - I removed them and added them directly to the server

`    resolve: {
        alias: {
            'eyepop': path.resolve(__dirname, '../../src/eyepop/dist/eyepop.min.js'),
            'eyepop-render-2d': path.resolve(__dirname, '../../src/eyepop-render-2d/dist/eyepop.render2d.min.js'),
            'example.jpg': path.resolve(__dirname, '../../examples/example.jpg'),
            'large_example.jpg': path.resolve(__dirname, '../../examples/large_example.jpg'),
        }
    },`

https://github.com/eyepop-ai/eyepop-sdk-node/pull/31/commits/1b14f6251cf82dc803abbcdb14a2710bb675b917

tschulz commented on examples/vite/src/vite-env.d.ts:
```diff
@@ -0,0 +1 @@
+/// <reference types="vite/client" />
```
is that an empty file?

64blit replied:
You are correct - has been deleted

64blit commented:
Added relative box padding and removed confidence label:

![image](https://github.com/eyepop-ai/eyepop-sdk-node/assets/61922101/84a274c7-d689-4db3-962c-8d21e84251e5)
